atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/FakeTransactionKeyValueServiceManager.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.transaction.impl;
+
+import com.palantir.atlasdb.cell.api.DdlManager;
+import com.palantir.atlasdb.cell.api.TransactionKeyValueService;
+import com.palantir.atlasdb.cell.api.TransactionKeyValueServiceManager;
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
+import com.palantir.logsafe.exceptions.SafeIllegalStateException;
+import java.util.Optional;
+import java.util.function.LongSupplier;
+
+public class FakeTransactionKeyValueServiceManager implements TransactionKeyValueServiceManager {
+    private final TransactionKeyValueService tkvsMock;
+
+    public FakeTransactionKeyValueServiceManager(TransactionKeyValueService tkvsMock) {
+        this.tkvsMock = tkvsMock;
+    }
+
+    @Override
+    public TransactionKeyValueService getTransactionKeyValueService(LongSupplier timestampSupplier) {
+        return tkvsMock;
```

Comment:
yeah I think we just extracted this so it's fine, but probably should just be named `delegate` or similar since it's passed in by whoever uses this